### PR TITLE
Add snaphot validation webhook

### PR DESCRIPTION
**Pull request by shaoyongfeng**
**How to categorize this PR?**
/area open-source
/kind enhancement
/platform alicloud

**What this PR does / why we need it**:
This PR adds the webhook to provide additional validation to volume snapshot objects i.e. VolumeSnapshot and VolumeSnapshotContent.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```other operator
The provider-alicloud extension now installs the external-snapshotter's validating webhook server for VolumeSnapshot and VolumeSnapshotContent objects. For more details check the corresponding [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook#kep-1900-add-additional-validation-to-volume-snapshot-objects).
```

### CODE REVIEW

File: charts/internal/seed-controlplane/charts/csi-alicloud/templates/csi-snapshot-validation-webhook-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{{- end }}
+        volumeMounts:
+          - name: csi-snapshot-validation
+            mountPath: /etc/csi-snapshot-validation
+            readOnly: true
+      volumes:
+        - name: csi-snapshot-validation
+          secret:
+            secretName: {{ .Values.csiSnapshotValidationWebhook.secrets.server }}
+
```
Comment (acumino) [inline charts/internal/seed-controlplane/charts/csi-alicloud/templates/csi-snapshot-validation-webhook-deployment.yaml:42]: Nit
```suggestion

```

File: pkg/controller/controlplane/valuesprovider.go [hunk 2]
```diff
@@ -55,6 +102,7 @@ func shootAccessSecretsFunc(namespace string) []*gutil.ShootAccessSecret {
 		gutil.NewShootAccessSecret("csi-snapshotter", namespace),
 		gutil.NewShootAccessSecret("csi-resizer", namespace),
 		gutil.NewShootAccessSecret("csi-snapshot-controller", namespace),
+		gutil.NewShootAccessSecret(alicloud.CSISnapshotValidation, namespace),
```
Comment (acumino) [inline pkg/controller/controlplane/valuesprovider.go:105]:
```suggestion
```

**Review by acumino — COMMENTED** [review]
Please add a similar check like this one - 
https://github.com/gardener/gardener-extension-provider-openstack/blob/1d8ffd8d1b983b9eae12a16d5502258374a3e242/pkg/controller/healthcheck/add.go#L95-L99

**Review by acumino — DISMISSED** [review]
/lgtm

**Review by EmoinLanyu — APPROVED** [review]
/lgtm

**Pull request merged**